### PR TITLE
16.0 web owl as odoo module jcb

### PR DESCRIPTION
**Pull request by caburj**
Create an odoo module for owl, so that using owl features and functions is done using the standard import syntax, instead of destructuring properties out of the owl global object

TASK-ID: 3032274

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/__manifest__.py [hunk 1]
```diff
@@ -87,6 +87,7 @@
             'web/static/lib/moment/moment.js',
             'web/static/lib/luxon/luxon.js',
             'web/static/lib/owl/owl.js',
+            'web/static/lib/owl/odoo_module.js',
```
Comment (caburj) [inline addons/web/__manifest__.py:90]: Note that `'web/static/lib/owl/*.js'` doesn't work because "odoo_module.js" is loaded before "owl.js".

**Comment by sdegueldre** [comment]
@robodoo rebase-ff

**Comment by caburj** [comment]
@sdegueldre I should remember doing that. I always forget.

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by sdegueldre** [comment]
> @sdegueldre I should remember doing that. I always forget.

Can't do that on your own PRs unless you have self-r+ rights

**Comment by hoangtrann** [comment]
I was pulling my hair today, as I setup a local instance using docker with Odoo image for version 16, along with enterprise addons, but it keeps saying `Missing dependencies:     ['@odoo/owl']`.

Turns out that this commit hasn't yet included in the weekly build (latest is Oct 25th), and I was checking enterprise repo at `HEAD`.

But as always, amazing work 💯

**Pull request closed without merging**